content/agent/autodiscovery/clusterchecks.md

```diff
@@ -0,0 +1,265 @@
+
+- By setting the `DD_EXTRA_CONFIG_PROVIDERS` environment variable:
+
+```
+DD_EXTRA_CONFIG_PROVIDERS="clusterchecks"
+```
+
+- Or adding it to your `datadog.yaml` configuration file:
+
+```
+config_providers:
+  - name: clusterchecks
+    polling: true
+```
+
+[Restart the Agent][5] to apply the configuration change.
+
+### Custom checks
+
+Running [custom Agent Checks][6] as Cluster Checks is supported, as long as all node-based Agents are able to run it. This means your check code:
+
+- must be installed on all node-based Agents where the `clusterchecks` config provider is enabled
+- must not depend on local resources that are not accessible to all agents
+
+## Setting up Check Configurations
+
+### Static configurations in files
+
+When the IP of a given resource is constant (eg. external service endpoint, public URL...), you can pass a static configuration to the Cluster Agent as yaml files. The file name convention and syntax are the sames as static configurations on the node-based Agent, with the addition of the `cluster_check: true` line.
+
+#### Example: MySQL check on CloudSQL database
+
+After setting up your CloudSQL instance and [setting up the datadog user][7], mount a `/conf.d/mysql.yaml` file in the Cluster Agent container with the following contents:
+
+```yaml
+cluster_check: true
+init_config:
+instances:
+  - server: '<PRIVATE_IP_ADDRESS>'
+    port: 3306
+    user: datadog
+    pass: '<YOUR_CHOSEN_PASSWORD>'
+```
+
+The `cluster_check` field will inform the Cluster Agent to delegate this check to one node-based Agent.
+
+### Template Source: Kubernetes Service Annotations
+
+As you can [annotate Kubernetes Pods][8], you can annotate Services with the following syntax:
+
+```yaml
+  ad.datadoghq.com/service.check_names: '[<CHECK_NAME>]'
+  ad.datadoghq.com/service.init_configs: '[<INIT_CONFIG>]'
+  ad.datadoghq.com/service.instances: '[<INSTANCE_CONFIG>]'
+```
+
+The `%%host%%` [template variable][9] is supported and will be replaced by the service's IP. The `kube_namespace` and `kube_service` tags will be added to the instance.
+
+#### Example: HTTP check on an nginx-backed service
+
+The following Service definition will expose the Pods from the `my-nginx` deployment and run an [HTTP check][10] to measure the latency of the load-balanced service:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: my-nginx
+  labels:
+    run: my-nginx
+  annotations:
+    ad.datadoghq.com/service.check_names: '["http_check"]'
+    ad.datadoghq.com/service.init_configs: '[{}]'
+    ad.datadoghq.com/service.instances: |
+      [
+        {
+          "name": "My Nginx",
+          "url": "http://%%host%%",
+          timeout: 1
+        }
+      ]
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: my-nginx
+```
+
+In addition, each pod should be monitored with the [NGINX check][11], as it allows to observe both each worker and the aggregated service.
```

Comment:
`both each worker and the aggregated service.` --> `each worker as well as the aggregated service`